modules/standard/Types.chpl

```diff
@@ -711,4 +711,110 @@ inline proc _bor_id(type t) return 0:t;
 pragma "no doc"
 inline proc _bxor_id(type t) return 0:t;
 
+
+/* Return the number of fields in a class or record as a param.
+   The count of fields includes types and param fields.
+ */
+proc numFields(type t) param
```

Comment:
OK, I've added the return types when not generic. I didn't do that originally because of the long-standing (but apparently fixed) bug where param functions didn't work if you specified a return type.

I don't know of any rule of thumb for ":returns:" fields - I tend to use them when the function documentation is complicated enough that it seems useful to describe the arguments separately and individually. These seem simple enough not to bother. Of course, I could write the whole thing in a ":returns:"... but that seems less nice to me.